sources/api/early-boot-config/src/provider/aws.rs

```diff
@@ -186,7 +189,16 @@ impl PlatformDataProvider for AwsDataProvider {
 
         let session_token = Self::fetch_imds_session_token(&client)?;
 
-        // Instance identity doc first, so the user has a chance to override
+        // Attempt to read from local file first on the `aws-dev` variant
+        #[cfg(bottlerocket_platform = "aws-dev")]
+        {
+            match local_file_user_data()? {
+                Some(s) => output.push(s),
+                None => warn!("No user data found via local file: {}", USER_DATA_FILE),
```

Comment:
I think not having a user data file is the normal case - is it worth a warning-level message?

Reply:
I think for variants that have the ability to use a local file it's worth having the message so users know that nothing was found there, similar to what we do for other sources.

Reply:
Can we have it as a info level message? I guess that would mean the message won't go to the console?

Reply:
I'll keep it as `warn!` for now to stay consistent with the rest of the program and ensure it outputs to the console as expected.